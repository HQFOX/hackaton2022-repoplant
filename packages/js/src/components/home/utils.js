import {
  Level1,
  Level2Average,
  Level3Bad,
  Level4,
  Level5
} from "@hv/uikit-react-icons";

export const getStatus = statusNumber => {
  switch (statusNumber) {
    case 1:
      return { Icon: Level1, sema: "sema10" };
    case 2:
      return { Icon: Level2Average, sema: "sema11" };
    case 3:
      return { Icon: Level3Bad, sema: "sema12" };
    case 4:
      return { Icon: Level4, sema: "sema13" };
    case 5:
      return { Icon: Level5, sema: "sema14" };
    default:
      return { Icon: null, sema: "sema1" };
  }
};
