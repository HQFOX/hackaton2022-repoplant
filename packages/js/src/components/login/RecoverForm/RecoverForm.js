import React, { useState } from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import {
  HvButton,
  HvTypography,
  HvLabel,
  HvBaseInput
} from "@hv/uikit-react-core";
import Message from "../Message";

const RecoverForm = ({ classes, onSubmit, onCancel, status }) => {
  const { t } = useTranslation();
  const [email, setEmail] = useState("");
  const isPending = status === "pending";
  const isSuccess = status === "success";
  const isError = status === "error";

  return (
    <div className={classes.root}>
      <HvTypography variant="mTitle" classes={{ root: classes.title }}>
        {t("pages.login.recoverForm.title")}
      </HvTypography>

      {(isError || isSuccess) && (
        <Message
          message={{
            error: t("pages.login.recoverForm.error"),
            success: t("pages.login.recoverForm.success")
          }}
          status={status}
        />
      )}

      <HvTypography variant="normalText" classes={{ root: classes.subtitle }}>
        {t("pages.login.recoverForm.subtitle")}
      </HvTypography>

      <HvLabel
        id="email-label"
        label={t("pages.login.recoverForm.email.label")}>
        <HvBaseInput
          id="email-input"
          placeholder={t("pages.login.recoverForm.email.placeholder")}
          value={email}
          inputProps={{ autoFocus: true }}
          onChange={evt => setEmail(evt.target.value)}
        />
      </HvLabel>

      <HvButton
        type="submit"
        category="ghost"
        className={classes.cancel}
        onClick={onCancel}>
        {t("pages.login.recoverForm.cancel")}
      </HvButton>

      <HvButton
        type="submit"
        category="primary"
        className={classes.recover}
        onClick={() => onSubmit(email)}
        disabled={isPending}>
        {isPending
          ? t("pages.login.recoverForm.submit.recovering")
          : t("pages.login.recoverForm.submit.recover")}
      </HvButton>
    </div>
  );
};

RecoverForm.propTypes = {
  /**
   * The classes object to be applied into the root object.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to root.
     */
    root: PropTypes.string.isRequired,
    /**
     * Styles applied to title.
     */
    title: PropTypes.string.isRequired,
    /**
     * Styles applied to subtitle.
     */
    subtitle: PropTypes.string.isRequired,
    /**
     * Styles applied to recover.
     */
    recover: PropTypes.string.isRequired,
    /**
     * Styles applied to cancel.
     */
    cancel: PropTypes.string.isRequired
  }).isRequired,
  /**
   * Callback to trigger on recover action.
   */
  onSubmit: PropTypes.func.isRequired,
  /**
   * Callback to trigger on cancel recover action.
   */
  onCancel: PropTypes.func.isRequired,
  /**
   * Recover status.
   */
  status: PropTypes.string.isRequired
};

export default RecoverForm;
