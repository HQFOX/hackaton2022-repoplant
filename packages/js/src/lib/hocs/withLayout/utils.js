export const findById = (data = [], id) => {
  for (let i = 0; i < data.length; i += 1) {
    const el = data[i];
    if (el.id === id) return el;

    const child = findById(el.data, id);
    if (child) return child;
  }
  return null;
};

export const findLeaf = item => (item.data ? findLeaf(item.data[0]) : item);

export const findFirstChildByParentId = (data = [], id) => {
  const parent = findById(data, id);
  return (parent && parent.data && findLeaf(parent.data[0]).id) || id;
};

export const navigationDepth = arr =>
  Array.isArray(arr)
    ? 1 + Math.max(...arr.map(el => navigationDepth(el.data)))
    : 0;
