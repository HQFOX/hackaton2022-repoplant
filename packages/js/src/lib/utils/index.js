export const getRandom = (max, min = 1) =>
  Math.floor(Math.random() * (max - min + 1)) + min;
