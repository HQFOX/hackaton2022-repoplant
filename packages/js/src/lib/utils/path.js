const getSelection = (data, pathname) => {
  let selection;

  data.forEach(item => {
    if (item.path === pathname) selection = { ...item };
    if (item.data) getSelection(item.data, pathname);
  });

  return selection;
};

export { getSelection };
