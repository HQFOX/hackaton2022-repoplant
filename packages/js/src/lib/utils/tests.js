import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { createMemoryHistory } from "history";
import HvProvider from "@hv/uikit-react-core/dist/Provider";

const mockStore = configureMockStore([thunk]);
const memoryHistory = createMemoryHistory();

const createMockStore = mockState => mockStore(mockState);

const withStoreProvider = (Component, store) => props => (
  <Provider store={createMockStore(store)}>
    <HvProvider>
      <MemoryRouter>
        <Component {...props} />
      </MemoryRouter>
    </HvProvider>
  </Provider>
);

const withHvProvider = Component => props => (
  <HvProvider>
    <MemoryRouter>
      <Component {...props} />
    </MemoryRouter>
  </HvProvider>
);

const withProps = Component => props => <Component {...props} />;

export {
  createMockStore,
  mockStore,
  memoryHistory,
  withStoreProvider,
  withHvProvider,
  withProps
};
