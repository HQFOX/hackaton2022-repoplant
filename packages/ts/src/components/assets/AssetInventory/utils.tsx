import React from "react";
import {
  AssetInventoryMetadata,
  AssetInventoryConfiguration,
  HvSemanticColorKeys
} from "@hv/uikit-react-core";
import {
  Delete,
  Preview,
  Upload,
  Level1,
  Level2Average,
  Level3Bad,
  Level4,
  Level5
} from "@hv/uikit-react-icons";

const metadata: AssetInventoryMetadata[] = [
  {
    id: "id1",
    accessor: "headerTitle",
    cellType: "alpha-numeric",
    searchable: true,
    sortable: true,
    sortableLabelAsc: "Title ascending",
    sortableLabelDesc: "Title descending"
  },
  {
    id: "id2",
    accessor: "semantic",
    cellType: "alpha-numeric"
  },
  {
    id: "id3",
    accessor: "probability",
    cellType: "numeric",
    searchable: true,
    sortable: true,
    sortableLabelAsc: "Probability ascending",
    sortableLabelDesc: "Probability descending"
  },
  {
    id: "id4",
    accessor: "timeHorizon",
    cellType: "numeric",
    sortable: true,
    sortableLabelAsc: "TimeHorizon ascending",
    sortableLabelDesc: "TimeHorizon descending"
  },
  {
    id: "id5",
    accessor: "event.schedule",
    cellType: "alpha-numeric",
    searchable: true
  },
  {
    id: "id6",
    accessor: "event.description",
    cellType: "alpha-numeric",
    searchable: true
  }
];

const viewConfiguration: AssetInventoryConfiguration = {
  breakpoints: { xs: 12, sm: 6, md: 4, lg: 3, xl: 3 },
  columnConfiguration: [
    { style: { width: 1, textAlign: "center" } },
    {
      title: "Event",
      style: { minWidth: "360px", textAlign: "start" }
    },
    {
      title: "Probability",
      style: { minWidth: "80px", textAlign: "end" }
    },
    {
      title: "Time horizon",
      style: { minWidth: "100px", textAlign: "end" }
    },
    {
      title: "Related Assets",
      style: {
        minWidth: "195px",
        paddingLeft: "30px",
        textAlign: "start"
      }
    }
  ]
};

const configuration = {
  metadata,
  viewConfiguration
};

const actions = [
  { id: "post", label: "Dismiss", disabled: false },
  {
    id: "get",
    label: "Preview",
    icon: <Preview color="atmo5" />,
    disabled: true
  },
  {
    id: "put",
    label: "Upload",
    icon: <Upload color="atmo5" />,
    disabled: true
  },
  { id: "delete", label: "Delete", icon: <Delete />, disabled: false }
];

type Status = {
  Icon: React.ComponentType<any> | null;
  sema: HvSemanticColorKeys;
};

const getStatus = (statusNumber: number): Status => {
  switch (statusNumber) {
    case 1:
      return { Icon: Level1, sema: "sema10" };
    case 2:
      return { Icon: Level2Average, sema: "sema11" };
    case 3:
      return { Icon: Level3Bad, sema: "sema12" };
    case 4:
      return { Icon: Level4, sema: "sema13" };
    case 5:
      return { Icon: Level5, sema: "sema14" };
    default:
      return { Icon: null, sema: "sema1" };
  }
};

export { configuration, actions, getStatus };
