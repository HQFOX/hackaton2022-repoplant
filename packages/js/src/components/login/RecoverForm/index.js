import { withStyles } from "@material-ui/core";
import styles from "./styles";
import RecoverForm from "./RecoverForm";

export default withStyles(styles, { name: "RecoverForm" })(RecoverForm);
