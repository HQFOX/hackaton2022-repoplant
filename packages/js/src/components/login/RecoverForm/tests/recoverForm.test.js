import React from "react";
import { mount } from "enzyme";
import { withHvProvider } from "lib/utils/tests";
import RecoverForm from "../index";

describe("<RecoverForm />", () => {
  let component;

  beforeEach(() => {
    const WithHvProvider = withHvProvider(RecoverForm);
    component = mount(<WithHvProvider login={jest.fn()} />);
  });

  it("it matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
