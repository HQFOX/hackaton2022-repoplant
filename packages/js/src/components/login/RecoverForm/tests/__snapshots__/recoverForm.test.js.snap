// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecoverForm /> it matches the snapshot 1`] = `undefined`;
